constants.h

```diff
@@ -20,38 +20,41 @@
 // set to "JQ6500" if using the cheap and friendly Aliexpress JQ6500
 #define HARDWARE_SOUNDBOARD_JQ6500 1
 
-// some inferior gate controllers don't BEEP at the correct pitch
-// frequence. Correct tones (in Hz) are below
+#define DELAY_INTERVAL_MS 60
 #define DELAY_DROP_TONE_MS 60
 #define DELAY_DROP_TONE_FINAL_MS 2250
 #define DELAY_GATE_RISE_WARN_MS 25
 #define DELAY_ABORT_TONE_1_MS 220
 #define DELAY_ABORT_TONE_2_MS 440
+#define GATE_ACTIVE_MS 300
 
 // random start min and max
 #define DELAY_RAND_MIN 100
 #define DELAY_RAND_MAX 2700
 
+// some inferior gate controllers don't BEEP at the correct pitch
+// frequence. Correct tones (in Hz) are below
 #define TONE_DROP_HZ 632
 #define TONE_GATE_RISE 1150
 #define TONE_ABORT_1_HZ 740
 #define TONE_ABORT_2_HZ 680
 #define TONE_VOLUME 130
 
 #define PIN_NEO_PIXEL 13 // light tree pin
-#define PIN_BUTTON_GO 2
-#define PIN_SPEAKER 11
-#define PIN_RELAY 8 // relay, or MOSFET
-#define PIN_LED_ACTIVE 5
-#define PIN_GATE_STATUS_LED_RED 18
-#define PIN_GATE_STATUS_LED_GREEN 19
-#define PIN_SFX_TX 10 // sound board transmit
-#define PIN_SFX_RX 9 // sound board receive
+#define PIN_BUTTON_GO 2  // The GO/ABORT button -interrupt 
+#define PIN_SPEAKER 11  // tone output
+#define PIN_RELAY 8 // relay, or MOSFET for GATE
+#define PIN_LED_ACTIVE 5  // On when GO button activated 
+#define PIN_GATE_STATUS_LED_RED 18 // Not used
+#define PIN_GATE_STATUS_LED_GREEN 19  //Not used
+#define PIN_SFX_TX 10 // sound board transmit (Arduino Rx)
+#define PIN_SFX_RX 9 // sound board receive (Arduino Tx)
+#define PIN_SENSOR 3 // light sensor - interrupt
 
 #define PIN_SFX_RST 0 // sound board reset
 #define PIN_SFX_ACT 1 // sound board active
 
-#define PIN_LIGHT_TREE_RELAY_ENABLE 3
+#define PIN_LIGHT_TREE_RELAY_ENABLE 4
```

Comment:
I know we have at least one bloke running a gate that expects to enable the light tree with pin 3. We might need to find another solution (we could use Pin 19).

Reply:
Pins 2 and 3 are the only ones that can be used for an interrupt input by my reading. 2 is used for ABORT so had to use 3 for the interrupt for the light beam

Reply:
argh! you're right. I might have to make light tree mode a software configuration item rather than a jumper.

Reply:
We should be able to listen for the beam while the sequence is running. As the sequence loop is still active, we can check at the start of each loop if the beam pin has gone low. As the ATMEGA is already busy running that loop, we won't run into any issues by doing the check there.

Reply:
Do you mean to not use an interrupt?  here are two examples of serial monitor using test button - first button is before sequence is finished - second is after finished.

[17430] No longer abortable
[17430] Set LED  1
[17550] Set LED  2
[17670] Set LED  3
[17790] Set LED  4
[17790] Gate Activate
[18091] Gate Deactivate
[20013] Reaction Time = 2223
[20013] Press GO to restart...
[20040] Setting sequence to STOP

[89785] No longer abortable
[89785] Set LED  1
[89905] Set LED  2
[90025] Set LED  3
[90145] Set LED  4
[90145] Gate Activate
[90445] Gate Deactivate
[92395] Setting sequence to STOP
[92395] Sequence complete - Waiting for reaction time...
[94418] Reaction Time = 4273
[94418] Press GO to restart...

Reply:
Yeah - should be do-able without an interrupt. We can keep the sequence loop running until we have a time. Pseudo code:
```
while (sequenceRunning || (timingEnabled && waitingForTime)) {
 if (timingEnabled && gateHasDropped) {
  if (digitalRead(PIN_TIMING_TRIGGER) == LOW) {
    // time found, save in var 
    waitingForTime = 0;
  }
 }
 // rest of sequence code
}
```